services/traction/api/endpoints/routes/v1/tenant/issuer/credentials.py

```diff
@@ -0,0 +1,130 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+from starlette import status
+
+from api.services.v1 import issuer_service
+from api.db.models.issue_credential import IssueCredentialRead
+
+
+from api.db.models.v1.contact import Contact
+
+from api.endpoints.dependencies.tenant_security import get_from_context
+from api.endpoints.dependencies.db import get_db
+
+
+from api.endpoints.models.tenant_workflow import (
+    TenantWorkflowStateType,
+)
+
+from api.endpoints.models.v1.issuer import (
+    CredentialsListResponse,
+    CredentialItem,
+    IssueCredentialPayload,
+    RevokeSchemaPayload,
+)
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+
+
+@router.get("/", response_model=CredentialsListResponse)
+async def get_issued_credentials(
+    state: TenantWorkflowStateType | None = None,
+    cred_issue_id: str | None = None,
+    db: AsyncSession = Depends(get_db),
+) -> CredentialsListResponse:
+    # this should take some query params, sorting and paging params...
+    wallet_id = get_from_context("TENANT_WALLET_ID")
+    tenant_id = get_from_context("TENANT_ID")
+
+    data = await issuer_service.get_issued_credentials(
+        db, tenant_id, wallet_id, None, cred_issue_id, state
+    )
+    resp_data = [
+        CredentialItem(
+            **d.__dict__,
+            contact_id=d.credential.contact_id,
+            credential_id=d.credential.id,
+            status="v0",  # v0
+            state=d.credential.issue_state,  # v0
+            created_at=d.workflow.created_at,
+            updated_at=d.workflow.updated_at,
+            alias="v0",
+        )
+        for d in data
+    ]
+
+    response = CredentialsListResponse(
+        items=resp_data, count=len(data), total=len(data)
+    )
+
+    return response
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED, response_model=CredentialItem)
+async def issue_new_credential(
+    payload: IssueCredentialPayload,
+    db: AsyncSession = Depends(get_db),
+) -> CredentialItem:
+    wallet_id = get_from_context("TENANT_WALLET_ID")
+    tenant_id = get_from_context("TENANT_ID")
+
+    # Use connection ID for v0 compatability.
+    contact = await Contact.get_by_id(
+        db,
+        tenant_id,
+        contact_id=payload.contact_id,
+        deleted=False,
+    )
+    data = await issuer_service.issue_new_credential(
+        db,
+        tenant_id,
+        wallet_id,
+        payload.cred_protocol,
+        payload.credential,
+        payload.cred_def_id,
+        contact.connection_id,
+        None,
+    )
+
+    response = CredentialItem(
+        **data.__dict__,
+        credential_id=data.credential.id,
+        status="v0",  # v0
+        state=data.credential.issue_state,  # v0
+        created_at=data.workflow.created_at,
+        updated_at=data.workflow.updated_at,
+        alias="v0",  # alias is none, CredentialItem won't allow none
+        contact_id=payload.contact_id,  # v0
+    )
+
+    return response
+
+
+@router.post(
+    "/revoke",
```

Comment:
we aren't using url parameters in other places... i don't oppose that being a url parameter, but i'm just worried it will be inconsistent. 

all of these are reasonable, not sure any of them are wrong. maybe there is a larger discussion there. 

`/contacts/<contact_id>/issue_credential` or 
`/issuer/credentials` or
`/governance/schemas/<schema_id>/issue`

Reply:
all 3 are valid, but you pick the one that makes the most sense... POST to the credentials collection makes sense, the you'd have the rest of the CRUD on the singular credential (ex. updating an external ref id, adding tags)... so you will end up with `GET /credentials/<id>`, `PUT /credentials/<id>` and then `POST /credentials/<id>/revoke` makes sense.